orttraining/orttraining/training_ops/cuda/communication/recv.cc

```diff
@@ -72,15 +73,15 @@ void Recv::ReceiveData(
   // count waiting time before setting up the actual communication.
   recvRange.Begin();
 #endif
-  buffer = AllocateBufferOnCPUPinned<char>(static_cast<size_t>(aggregated_aligned_tensor_bytes));
+  buffer = GetScratchBuffer<char>(aggregated_aligned_tensor_bytes);
```

Comment:
>GetScratchBuffer [](start = 11, length = 16)

note that ScratchBuffer is intended to use in default stream, so it's nature to have nccl running in default stream too.

Reply:
NCCL service invokes all NCCL APIs using default stream now.

---
In reply to: [483803320](https://github.com/microsoft/onnxruntime/pull/5054#discussion_r483803320) [](ancestors = 483803320)